From af6a70a816d5a98d22b5bb1012e21cc2ecb4d5db Mon Sep 17 00:00:00 2001
From: Peter Chiu <chui-hao.chiu@mediatek.com>
Date: Thu, 4 Aug 2022 14:08:11 +0800
Subject: [PATCH] mt76: mt7915: reduce TWT SP sent to FW for cert

Set TWT SP duration to 88 percent to prevent HW sends PPDU over TWT SP.

Signed-off-by: Peter Chiu <chui-hao.chiu@mediatek.com>
---
 mt7915/mcu.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mt7915/mcu.c b/mt7915/mcu.c
index d7e717f4..2bd67f7c 100644
--- a/mt7915/mcu.c
+++ b/mt7915/mcu.c
@@ -3538,7 +3538,7 @@ int mt7915_mcu_twt_agrt_update(struct mt7915_dev *dev,
 		.own_mac_idx = mvif->mt76.omac_idx,
 		.flowid = flow->id,
 		.peer_id = cpu_to_le16(flow->wcid),
-		.duration = flow->duration,
+		.duration = (flow->duration * 7) >> 3,
 		.bss_idx = mvif->mt76.idx,
 		.start_tsf = cpu_to_le64(flow->tsf),
 		.mantissa = flow->mantissa,
-- 
2.18.0

